devices/ePaper/Drivers/Ssd168x/Ssd1681/Ssd1681.cs

```diff
@@ -0,0 +1,295 @@
+            if (CurrentFrameBufferPageLowerBound <= frameByteIndex
+                && frameByteIndex < CurrentFrameBufferPageUpperBound)
+            {
+                /*
+                 * Lookup Table for colors on SSD1681
+                 * 
+                 *  LUT for Red, Black, and White ePaper display
+                 * 
+                 * |                |               |                    |
+                 * | Data Red RAM   | Data B/W RAM  | Result Pixel Color |
+                 * |----------------|---------------|--------------------|
+                 * |        0       |       0       |       Black        |
+                 * |        0       |       1       |       White        |
+                 * |        1       |       0       |       Red          |
+                 * |        1       |       1       |       Red          |
+                 * 
+                 * 
+                 *  LUT for Black and White ePaper display with SSD1681
+                 * |                |               |                    |
+                 * | Data Red RAM   | Data B/W RAM  | Result Pixel Color |
+                 * |----------------|---------------|--------------------|
+                 * |        0       |       0       |       Black        |
+                 * |        0       |       1       |       White        |
+                 * |        1       |       0       |       Black        |
+                 * |        1       |       1       |       White        |
+                 */
+
+                if (color.R >= 128 && color.G == 0 && color.B == 0)
+                {
+                    // this is a colored pixel
+                    // according to LUT, no need to change the pixel value in B/W buffer.
+                    // red frame buffer starts with 0x00. ORing it to set red pixel to 1.
+                    _frameBuffer2bpp.ColorBuffer[pageByteIndex] |= (byte)(128 >> (x & 7));
+                }
+                else if (color.R == 0 && color.G == 0 && color.B == 0)
+                {
+                    // black pixel
+                    _frameBuffer2bpp.BlackBuffer[pageByteIndex] &= (byte)~(128 >> (x & 7));
+                }
+                else
+                {
+                    // assume white if R, G, and B > 0
+                    _frameBuffer2bpp.BlackBuffer[pageByteIndex] |= (byte)(128 >> (x & 7));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Draws the specified buffer directly to the Black/White RAM on the display.
+        /// Call <see cref="Ssd168x.PerformFullRefresh"/> after to update the display.
+        /// </summary>
+        /// <param name="startXPos">X start position.</param>
+        /// <param name="startYPos">Y start position.</param>
+        /// <param name="buffer">The buffer array to draw.</param>
+        public void DirectDrawBuffer(int startXPos, int startYPos, params byte[] buffer)
```

Comment:
same for this functions and few others. So I really may have missed something

Reply:
Yup, you're right. My original thinking was that there might be other SSD168x controllers with their own initialization sequence and buffer requirements which is why I didn't have those in the base-class. I did a bit more digging on the SSD168x line of controllers and seems like there is the SSD1680(A), SSD1681, and SSD1683 (which I don't have). I skimmed through the SSD1683 datasheet and looks like it uses the same commands and has the same initialization sequence, buffer, etc... . I have moved all the code into the base class now.